### ISSUE TITLE
nwnsc not terminating after done compiling

### ISSUE THREAD

**Issue by StefanoND** [nwneetools/nwnsc#41]
Hello, first of all thanks for nwnsc.

I'm using Linux, Manjaro 6.3.5-2 and VSCodium (with marketplace and unblocked features)

I've been using nwnsc with VSCode for a while now and I've always noticed that as time goes on, my computer fans starts ramping up and the entire system starts to slow down to a point where I'm forced to reboot to fix this issue.

Then one day I decided to look into what was causing it, I ran "htop" command in the terminal to check for running processes and noticed that there was a lot of nwnsc instances running.

Since I'm too lazy I just created a script that runs "sudo killall nwnsc" periodically instead of posting the issue here.

So basically everytime nwnsc runs it never terminates it after its done.

Here's a screen snip of my htop

PS: I've already posted an issue in https://github.com/PhilippeChab/nwscript-ee-language-server which is the plugin that I'm using to run it, and the dev said that the issue is not with his plugin but with nwnsc itself (exclusively on linux systems), so that's why I'm posting it here.

Thanks again

![image](https://github.com/nwneetools/nwnsc/assets/45202623/059832e6-da11-46f0-ac16-f874e504c63f)

**Comment by jakkn** [comment]
Hi @StefanoND, thank you for the bug report! I can confirm that this happens, and it seems to trigger when something fails to compile.

To reproduce you just need any syntax error. Example:
```nss
void main() {
  return
}
```

I tested a couple different versions and found that [v1.1.3](https://github.com/nwneetools/nwnsc/releases/tag/v1.1.3) works, while [v1.1.4](https://github.com/nwneetools/nwnsc/releases/tag/v1.1.4) doesn't.
I also looked over the diff https://github.com/nwneetools/nwnsc/compare/v1.1.3...v1.1.4 but nothing stands out to my untrained eye as breaking.

The next step would be to try to bisect the error, but this project doesn't compile on my system so I'm not going to be of immediate help here. Perhaps @niv or @PhilippeChab can lend a hand as the contributors on v1.1.4 :pray:

**Comment by PhilippeChab** [comment]
The problematic commit would be this [one](https://github.com/nwneetools/nwnsc/commit/4e8351eb81f7b694774d18e060e69d1e11f26be7)?

**Comment by PhilippeChab** [comment]
Typo [here](https://github.com/nwneetools/nwnsc/blob/36bb6d89b223a8aea98fd42b58d405035b7f1760/_NscLib/NscCodeGenerator.h#L101) might be the problem?
